main.py

```diff
@@ -177,82 +122,120 @@ def noCode():
         pixels[noCode]=(0, 0, 0)
     pixels.show()
     time.sleep(0.3)
-#while True:
-    #noCode()
 
-def disabled():
-    for disabled in range(0, 15):
-        pixels[disabled]=(255, 0, 0)
+def RobotDisabled(current_mode):
+    for countdown in range(steps, 1, -1):
+        if read_current_mode() != current_mode:
+            return
+        for i in range(20): #disabled takes priority over conecube mode
+            pixels[i]=(255*countdown/steps, 0, 0)
         pixels.show()
-    for alliance in range(15, 20):
-        pixels[alliance]=(allianceColor)
-        pixels.show
-#while True:
-    #disabled()
-
-#new
-temp = pin13.value
-temp2 = pin12.value<<1
-temp3 = pin11.value<<2
-temp4 = pin10.value<<3
-temp5 = pin9.value<<4
-
-temp_total = (temp+temp2+temp3+temp4+temp5)
-print(temp_total)
-
-#pin7value
-print(pin7.value)
+        time.sleep(wait)
+    for countup in range(1, steps, 1):
+        read_conecube_mode()
+        if read_current_mode() != current_mode:
+            return
+        for i in range(20):
+            pixels[i]=(255*countup/steps, 0, 0)
+        pixels.show()
+        time.sleep(wait)
 
-#pin9value
-print(pin9.value)
+def arm_Position():
+    for i in range((20-conecube_length)):
+        pixels[i]=(0, 255, 0)
+    pixels.show()
 
-#pin10value
-print(pin10.value)
+def display_Cube():
+    for i in range((20-conecube_length)):
+        pixels[i]=(145, 0, 255)
+    pixels.show()
 
-#pin11value
-print(pin11.value)
+def display_Cone():
+    for i in range((20-conecube_length)):
+        pixels[i]=(255, 165, 0)
+    pixels.show()
 
-#pin12value
-print(pin12.value)
+def blinkingCube():
+    for cube in range((20-conecube_length)):
+        pixels[cube]=(145, 0, 255)
+    pixels.show()
+    time.sleep(0.3)
+    for cube in range((20-conecube_length)):
+        pixels[cube]=(0, 0, 0)
+    pixels.show()
+    time.sleep(0.2)
 
-#pin13value
-print(pin13.value)
+def blinkingCone():
+    for cone in range((20-conecube_length)):
+        pixels[cone]=(255, 165, 0)
+    pixels.show()
+    time.sleep(0.3)
+    for cone in range((20-conecube_length)):
+        pixels[cone]=(0, 0, 0)
+    pixels.show()
+    time.sleep(0.2)
 
-time.sleep(5)
+def display_conecube(color): #does not have a show for fast execution
+    for i in range(20-conecube_length, 20, 1):
+        pixels[i]=color
+
+def GamePiece_Collected():    
```

Comment:
nit: python function names should start with lower case letters and be snake_case